File: grizgo-web/src/app/restaurants/models/menu.ts
export interface MenuItemSummary {
  id: string;
  nameSr: string;
  descriptionSr: string;
  price: number;
  imageUrl: string;
  isAvailable: boolean;
}

export interface MenuCategory {
  id: string;
  nameSr: string;
  displayOrder: number;
  items: MenuItemSummary[];
}

export interface Menu {
  menuId: string;
  nameSr: string;
  categories: MenuCategory[];
}

export interface RestaurantMenuList {
  restaurantId: string;
  menus: Menu[];
}

export interface CreateMenuRequest {
  nameSr: string;
  nameEn: string;
  descriptionSr: string;
  descriptionEn: string;
  displayOrder: number;
}

export interface CreateMenuCategoryRequest {
  nameSr: string;
  nameEn: string;
  descriptionSr: string;
  descriptionEn: string;
  displayOrder: number;
}

export interface CreateMenuItemRequest {
  nameSr: string;
  nameEn: string;
  descriptionSr: string;
  descriptionEn: string;
  price: number;
  imageUrl: string;
  isAvailable: boolean;
  isFeatured: boolean;
  preparationTimeMinutes: number;
}

File: RestaurantService.Domain/Entities/Menu.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RestaurantService.Domain.Entities
{
    public class Menu
    {
        public Guid Id { get; set; }

        public Guid RestaurantId { get; set; }

        public string NameSr { get; set; } = string.Empty;

        public string NameEn { get; set; } = string.Empty;

        public string DescriptionSr { get; set; } = string.Empty;

        public string DescriptionEn { get; set; } = string.Empty;

        public bool IsActive { get; set; } = true;

        public int DisplayOrder { get; set; }

        public Restaurant Restaurant { get; set; } = null!;

        public List<MenuCategory> Categories { get; set; } = new();

        public List<Promotion> Promotions { get; set; } = new();
    }
}
